aws/logs_monitoring/parsing.py

```diff
@@ -528,6 +528,116 @@ def cwevent_handler(event, metadata):
     yield data
 
 
+def parse_aws_waf_logs(event):
+    """Parse out complex arrays of objects in AWS WAF logs
+
+    Attributes to convert:
+        httpRequest.headers
+        nonTerminatingMatchingRules
+        rateBasedRuleList
+        ruleGroupList
+
+    This prevents having an unparsable array of objects in the final log.
+    """
+    if event.get(DD_SOURCE) != "waf":
+        return event
+    event_copy = copy.deepcopy(event)
+
+    message = event_copy.get("message", {})
+    if isinstance(message, str):
+        try:
+            message = json.loads(message)
+        except json.JSONDecodeError:
+            logger.debug("Failed to decode waf message")
+            return
+
+    headers = message.get("httpRequest", {}).get("headers")
+    if headers:
+        message["httpRequest"]["headers"] = {}
+        for index, item in enumerate(headers):
+            current_header = headers[index]
+            name = current_header["name"]
+            value = current_header["value"]
+            message["httpRequest"]["headers"][name] = value
+
+    # Iterate through rules in ruleGroupList and nest them under the group id
+    # ruleGroupList has three attributes that need to be handled separately
+    rule_groups = message.get("ruleGroupList", {})
+    if rule_groups:
+        message["ruleGroupList"] = {}
+        for index, item in enumerate(rule_groups):
+            current_rule_group = rule_groups[index]
+            group_id = current_rule_group.pop("ruleGroupId", None)
+            if not group_id in message["ruleGroupList"]:
```

Comment:
If `group_id` is `None` from the previous statement, this will add an element with key `None` to `message["ruleGroupList"]`. We should update this check (and other similar checks) to make sure `group_id` is not `None`.